docs/pages/api/index.md

```diff
@@ -12,33 +12,115 @@ env = gym.make('CartPole-v0')
+we want to reset the environment to a new initial state. The environment issues a done signal to the agent if it enters such a terminal state.
+Not all done signals must be triggered by a "catastrophic failure": Sometimes we also want to issue a done signal after
+a fixed number of timesteps, or if the agent has succeeded in completing some task in the environment.
+
+Let's see what the agent-environment loop looks like in Gym.
+This example will run an instance of `LunarLander-v2` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the environment
 
 ```python
 import gym
-env = gym.make('CartPole-v0')
-env.reset()
-for _ in range(1000): 
-	env.render()  # by default `mode="human"`(GUI), you can pass `mode="rgb_array"` to retrieve an image instead
-	env.step(env.action_space.sample())  # take a random action 	
+env = gym.make("LunarLander-v2")
+env.action_space.seed(42)
+
+observation, info = env.reset(seed=42, return_info=True)
+
+for _ in range(1000):
+    env.render()
+    observation, reward, done, info = env.step(env.action_space.sample())
+
+    if done:
+      observation, info = env.reset(return_info=True)
+
 env.close()
 ```
 
 The output should look something like this
 
-![cartpole-no-reset](https://user-images.githubusercontent.com/28860173/129241283-70069f7c-453d-4670-a226-854203bd8a1b.gif)
+![lunar_lander](https://user-images.githubusercontent.com/15806078/153222406-af5ce6f0-4696-4a24-a683-46ad4939170c.gif)
+
+
+## Commonly used methods
+>### Stepping
+> `step(action: ActType) -> Tuple[ObsType, float, bool, dict]`
+> 
+>Run one timestep of the environment's dynamics. When end of episode is reached, you are responsible for calling `reset()`
+to reset this environment's state. Accepts an action and returns a tuple `(observation, reward, done, info)`.
+>
+>>**Parameters**
+>>- `action`(**object**): an action provided by the agent. This will oftentimes be a numpy array, but might also be 
+>> an integer (for discrete action spaces) or a more complex object.
+> 
+>>**Returns**
+>>
+>> This method returns a tuple `(observation, reward, done, info)`:
+>>- `observation` (**object**): agent's observation of the current environment.
+>> This may, for instance, be a numpy array containing the positions and velocities of certain objects.
+>>- `reward` (**float**) : amount of reward returned after previous action
+>>- `done` (**bool**): whether the episode has ended, in which case further `step()` calls will return undefined results.
+>> A done signal may be emitted for different reasons: Maybe the task underlying the environment was solved successfully, 
+>> a certain timelimit was exceeded, or the physics simulation has entered an invalid state. `info` may contain additional
+>> information regarding the reasone for a done signal.
+>>- `info` (**dict**): contains auxiliary diagnostic information (helpful for debugging, and sometimes learning)
```

Comment:
I'd also mention logging metrics

Reply:
Also copied from a docstring. I will make a corresponding change to that docstring. However, I'm not completely sure what you mean by metrics. We are not talking about losses, right? Could you give an example?

Reply:
As an example from the top of my mind - I'm training some agents to move from point A to point B, so in the info dict I pass the current distance to the destination; then I log that information and use it to track the training progress, since it's a measure of the agent's performance (i.e. a metric)